app/Filament/Resources/UserResource/Pages/ListUsers.php

```diff
@@ -154,6 +157,14 @@ public function table(Table $table): Table
                     ->multiple()
                     ->searchable()
                     ->preload(),
+                SelectFilter::make('roles')
+                    ->label('Roles')
+                    ->options($this->getRolesOption())
```

Comment:
Personally I think the function should be passed to `options()` instead of defined in the class (`options(fn () => [.......])`) so that the filter is more portable, otherwise you need to copy both